mmv1/products/discoveryengine/SearchEngine.yaml

```diff
@@ -0,0 +1,147 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'SearchEngine'
+description: |
+  Vertex AI Search and Conversation can be used to create a search engine or a chat application by connecting it with a datastore
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Create a Search Engine': 'https://cloud.google.com/generative-ai-app-builder/docs/create-engine-es'
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.engines'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines?engineId={{engine_id}}'
+delete_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+
+update_verb: :PATCH
+update_mask: true
+
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}',
+  ]
+
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "discoveryengine_searchengine_basic"
+    primary_resource_id: 'basic'
+    primary_resource_name:
+      'fmt.Sprintf("tf_test_search_engine%s", context["random_suffix"])'
+    vars:
+      engine_id: "example-engine-id"
+      data_store_id: "example-datastore-id"
+
+parameters:
```

Comment:
For `required` and `immutable` attribute, we have default value for both, so we can remove `required: false` and `immutable: false` for the fields in this file.

Reply:
Do you mean that we have default values for  engine_id and datastore_id?

Reply:
sorry, I was not clear in my previous comments. `required` and `immutable` attribute for fields are set to `false` by default, so we can omit `required: false` for any optional fields and `immutable: false` for updatable fields

Reply:
Will address these changes in the new PR